src/models/ComponentStyle.js

```diff
@@ -36,15 +34,17 @@ export default (nameGenerator: NameGenerator) => {
      * Returns the hash to be injected on render()
      * */
     generateAndInjectStyles(executionContext: Object) {
-      const flatCSS = flatten(this.rules, executionContext).join('')
-        .replace(/^\s*\/\/.*$/gm, '') // replace JS comments
-      const hash = hashStr(this.componentId + flatCSS)
+      const flatCSS = flatten(this.rules, executionContext)
+      const hash = hashStr(this.componentId + flatCSS.join(''))
+
```

Comment:
I am just pointing to a little bug. "inserted[hash]" must be a global variable for Styled, as long "result" works in global scope. Not sure is this related to this PR, but it is better to change it. Someday.
<img width="312" alt="screen shot 2017-03-18 at 9 55 59 am" src="https://cloud.githubusercontent.com/assets/582410/24065698/32d8dbfe-0bc1-11e7-9686-5d5ac30efcbb.png">